File: mock_llm_service.py
import sys
from http.server import BaseHTTPRequestHandler, HTTPServer
import json

class Handler(BaseHTTPRequestHandler):
    def do_POST(self):
        response = {
            "score": self.server.score,
            "label": self.server.label
        }
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps(response).encode())

def run(label, port):
    if label == "left":
        score = -1.0
    elif label == "center":
        score = 0.0
    elif label == "right":
        score = 1.0
    else:
        score = 0.0

    server = HTTPServer(('', port), Handler)
    server.label = label
    server.score = score
    print(f"Starting mock LLM service for {label} on port {port}...")
    server.serve_forever()

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python mock_llm_service.py <label> <port>")
        sys.exit(1)
    label = sys.argv[1]
    port = int(sys.argv[2])
    run(label, port)

File: mock_llm_service.go
package main

import (
	"encoding/json"
	"log"
	"net/http"
	"os"
)

type AnalysisResult struct {
	Score float64 `json:"score"`
	Label string  `json:"label"`
}

func analyzeHandler(label string, score float64) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		result := AnalysisResult{
			Score: score,
			Label: label,
		}
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(result)
	}
}

func main() {
	if len(os.Args) < 3 {
		log.Fatal("Usage: mock_llm_service <label> <port>")
	}
	label := os.Args[1]
	port := os.Args[2]

	var score float64
	switch label {
	case "left":
		score = -1.0
	case "center":
		score = 0.0
	case "right":
		score = 1.0
	default:
		score = 0.0
	}

	http.HandleFunc("/analyze", analyzeHandler(label, score))

	log.Printf("Starting mock LLM service for %s on port %s...", label, port)
	err := http.ListenAndServe(":"+port, nil)
	if err != nil {
		log.Fatalf("Failed to start server: %v", err)
	}
}